### PR TITLE
Generate fields as non-pub if they would be access restricted in C++.

### LINKED ISSUE #1926: Private C++ members generate bindings that are public.
### Input C/C++ Header

```C++
class PubPriv {
  public:
    int x;
  private:
    int y;
};
```

### Bindgen Invocation

```rust
bindgen::Builder::default()
    .header("input.h")
    .clang_args(&["-x", "c++"])
    .generate()
    .unwrap()
```

### Actual Results

```rust
#[repr(C)]
pub struct PubPriv {
    pub x: ::std::os::raw::c_int,
    pub y: ::std::os::raw::c_int,
}
```

### Expected Results

```rust
#[repr(C)]
pub struct PubPriv {
    pub x: ::std::os::raw::c_int,
    y: ::std::os::raw::c_int,
}
```

Declaring `y` as `pub` lets Rust read `y` and construct a `PubPriv` when neither should really be allowed, as well as making it hard for tools to treat the two fields differently, like [autocxx issue 53](https://github.com/google/autocxx/issues/53) wants to do.

### PR DESCRIPTION
**Pull request by SilensAngelusNex**
Fixes #1926

### CODE REVIEW

File: src/ir/comp.rs [hunk 1]
```diff
@@ -1280,6 +1309,8 @@ impl CompInfo {
                     let comment = cur.raw_comment();
                     let annotations = Annotations::new(&cur);
                     let name = cur.spelling();
+                    let is_public =
+                        cur.access_specifier() != clang_sys::CX_CXXPrivate;
```
Comment (emilio) [inline src/ir/comp.rs:1313]: Why this? The handling of `protected` fields seems inconsistent with the code below.

File: src/ir/comp.rs [hunk 2]
```diff
@@ -1353,9 +1385,14 @@ impl CompInfo {
                         cur.kind() != CXCursor_EnumDecl
                     {
                         let ty = cur.cur_type();
+                        let public = cur.access_specifier() ==
+                            clang_sys::CX_CXXPublic ||
```
Comment (emilio) [inline src/ir/comp.rs:1389]: please call access_specifier() just once. It's an ffi call after all so it's not like the compiler can remove redundant calls to it.

File: tests/expectations/tests/class_nested.rs [hunk 3]
```diff
@@ -77,7 +77,7 @@ fn bindgen_test_layout_A() {
 #[repr(C)]
 #[derive(Debug, Default, Copy, Clone, Hash, PartialEq, Eq)]
 pub struct A_C {
-    pub baz: ::std::os::raw::c_int,
+    baz: ::std::os::raw::c_int,
```
Comment (emilio) [inline tests/expectations/tests/class_nested.rs:80]: This will break many consumers, I think we should put it behind an opt-in flag, wdyt?

Comment (SilensAngelusNex) [reply tests/expectations/tests/class_nested.rs]: SGTM; done.

File: src/options.rs [hunk 4]
```diff
@@ -915,6 +918,11 @@ where
         builder = builder.dynamic_library_name(dynamic_library_name);
     }
 
+    if matches.is_present("respect-cxx-access-specs") {
+        println!("Respecting private items.");
```
Comment (emilio) [inline src/options.rs:922]: Please remove this?

**Review by emilio — APPROVED** [review]
Looks good, thanks! Can you squash your commits and remove the stray println? Thanks!

File: src/options.rs (hunk 4 above)
Comment (SilensAngelusNex) [reply src/options.rs]: Oops, nice catch!

**Comment by SilensAngelusNex** [comment]
This isn't waiting on anything more from me is it?

**Comment by emilio** [comment]
(No, not waiting for you sorry, was just swamped with exams and work)

**Review by emilio — APPROVED** [review]
Still looks good :)

**Pull request merged**